src/repo.rs

```diff
@@ -0,0 +1,101 @@
+use std::env;
+use std::io::{self, Error, ErrorKind};
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+
+use git2::{build::RepoBuilder, Cred, FetchOptions, RemoteCallbacks};
+use tempfile::TempDir;
+
+pub struct RepoManager {
+    pub temp_dir: TempDir,
+}
+
+impl RepoManager {
+    pub fn new() -> Result<Self, io::Error> {
+        TempDir::new()
+            .map(|temp_dir| Self { temp_dir })
+            .map_err(|_| Error::new(ErrorKind::Other, "Failed to create temp dir"))
+    }
+
+    pub fn clone_repo(&self, repo_url: &str, dest_name: &str) -> Result<PathBuf, git2::Error> {
+        let dest_path = self.temp_dir.path().join(dest_name);
+
+        let mut callbacks = RemoteCallbacks::new();
+        let mut attempted = false;
+
+        callbacks.credentials(move |_url, username_from_url, _allowed_types| {
+            if attempted {
+                return Err(git2::Error::from_str("❌ SSH authentication failed"));
+            }
+            attempted = true;
+
+            match username_from_url {
+                Some(username) => Cred::ssh_key_from_agent(username),
+                None => Err(git2::Error::from_str(
+                    "❌ Username for SSH authentication is missing",
+                )),
+            }
+        });
+
+        let mut fetch_options = FetchOptions::new();
+        fetch_options.remote_callbacks(callbacks);
+
+        let mut builder = RepoBuilder::new();
+        builder.fetch_options(fetch_options);
+
+        match builder.clone(repo_url, &dest_path) {
+            Ok(_) => {
+                println!("✅ Successfully cloned {repo_url} into {dest_path:?}");
```

Comment:
Logging the path to the temporary folder doesn't make sense. could you please remove path value?